plugins/analytics/public/github.py

```diff
@@ -0,0 +1,158 @@
+    """
+
+    _defaults = {
+        "name": "GithubMonitor",
+        "description": "Executes Github queries (stored as indicators) and create relevant observables.",
+        "frequency": timedelta(hours=24),
+    }
+
+    def _get_or_create_observable(self, obs_type, value):
+        cls = observable.TYPE_MAPPING[obs_type]
+        obs = cls.find(value=value)
+        if not obs:
+            obs = cls(value=value).save()
+        return obs
+
+    def create_code_observable(self, code, query, tags):
+        logging.info(f"Processing code: {code.path}")
+        md5 = hashlib.md5(code.decoded_content).hexdigest()
+        sha1 = hashlib.sha1(code.decoded_content).hexdigest()
+        sha256 = hashlib.sha256(code.decoded_content).hexdigest()
+        value = f"{code.path}:{sha256}"
+        context = {
+            "content": f"{code.decoded_content.decode('utf-8')}",
+            "path": code.path,
+            "url": code.url,
+            "download_url": code.download_url,
+        }
+        code_obs = self._get_or_create_observable(ObservableType.file, value)
+        code_obs.md5 = md5
+        code_obs.sha1 = hashlib.sha1(code.decoded_content).hexdigest()
+        code_obs.sha256 = sha256
+        code_obs.size = code.size
+        code_obs.tag(tags + ["github-code"])
+        code_obs.add_context(self.name, context)
+        code_obs.save()
+        md5_obs = self._get_or_create_observable(ObservableType.md5, md5)
+        sha1_obs = self._get_or_create_observable(ObservableType.sha1, sha1)
+        sha256_obs = self._get_or_create_observable(ObservableType.sha256, sha256)
+        md5_obs.tag(tags + ["github-code"])
+        sha1_obs.tag(tags + ["github-code"])
+        sha256_obs.tag(tags + ["github-code"])
+        code_obs.link_to(md5_obs, "md5", "")
+        code_obs.link_to(sha1_obs, "sha1", "")
+        code_obs.link_to(sha256_obs, "sha256", "")
+        return code_obs.save()
+
+    def create_repository_observable(self, repository, query, tags):
+        logging.info(f"Processing repository: {repository.html_url}")
+        repository_obs = self._get_or_create_observable(
+            ObservableType.url, repository.html_url
+        )
+        repository_obs.tag(tags + ["github-repository"])
+        context = {
+            "description": repository.description,
+            "branches": [branch.name for branch in repository.get_branches()],
+            "id": repository.id,
+            "fullname": repository.full_name,
+            "last modified": repository.last_modified,
+            "pushed at": repository.pushed_at,
+            "created at": repository.created_at,
+            "stars": repository.stargazers_count,
+            "subscribers": repository.subscribers_count,
+            "forks": repository.forks_count,
+            "watchers": repository.watchers_count,
+            "matching query": query,
+        }
+        repository_obs.add_context(self.name, context)
```

Comment:
Do you want to keep track of last-modified times (and watchers, and forks, and stars)? if not, you ca use `skip_compare` which will skip the `last modified` field when doing the comparison to determine whether it's the same context or not.

Reply:
Yes, makes total sense!

Reply:
I will update this implementation after merging a PR to support `add_context(overwrite=True)`

Reply:
Updated and overwrite existing context.